sync_issues_to_jira/push_event.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+#
+# Copyright 2019 Espressif Systems (Shanghai) PTE LTD
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from github import Github
+import os
+import re
+
+def handle_push_event(event):
+    issue_numbers = []
+    for commit in event['commits']:
+        commit_message = commit['message']
+        issue_numbers += parse_commit_message(commit_message)
+    
+    github = Github(os.environ['GITHUB_TOKEN'])
+    repo = github.get_repo(os.environ['GITHUB_REPOSITORY'])
+    for issue in issue_numbers:
+        gh_issue = repo.get_issue(int(issue))
+        if gh_issue.pull_request:
+            update_pull_request(gh_issue.as_pull_request())
+
+def update_pull_request(pull_request):
+    print("Updating %s" % pull_request)
+    if pull_request.state == 'open':
+        print('Pull request is open, nothing to update.')
+        return
+    original_title = pull_request.title
+    # Prepend [Merged] to the pull request title
+    new_title = '[Merged] ' + original_title
+    pull_request.edit(title=new_title)
+    # Thank contributor for opening pull request. Let them know we didn't throw it away
+    pull_request.create_issue_comment('The pull request has been cherry-picked, the commit is linked above.\
```

Comment:
Sometimes we actually do "close" a PR by doing something different to what the author originally intended, and with a totally different commit. We try to avoid this, but it's necessary sometimes.

Ideally this action would check if the commit author is the same as the author of any of the commits in the PR branch, and only continue if it is. However this is probably a rare enough occasion that you can consider it a "nice to have". :)

Reply:
Thank you for suggestion. I have noted this feature request and will implement it in the future.